gcp_airflow_foundations/base_class/source_template_config.py

```diff
@@ -0,0 +1,152 @@
+from airflow.exceptions import AirflowException
+
+from dacite import Config
+from dataclasses import dataclass, field
+from dacite import from_dict
+
+
+from pydantic import validator, root_validator
+
+import datetime
+import re
+from typing import List, Optional
+
+from gcp_airflow_foundations.base_class.ods_metadata_config import (
+    OdsTableMetadataConfig,
+)
+from gcp_airflow_foundations.enums.ingestion_type import IngestionType
+from gcp_airflow_foundations.enums.hds_table_type import HdsTableType
+
+from gcp_airflow_foundations.base_class.ods_table_config import OdsTableConfig
+from gcp_airflow_foundations.base_class.hds_table_config import HdsTableConfig
+from gcp_airflow_foundations.base_class.facebook_table_config import FacebookTableConfig
+from gcp_airflow_foundations.base_class.template_ingestion_options_config import TemplateIngestionOptionsConfig
+from gcp_airflow_foundations.base_class.column_udf_config import ColumnUDFConfig
+
+
+@dataclass
+class SourceTemplateConfig:
```

Comment:
If you want, you could create a base class that both SourceTemplateConfig and TableConfig inherit from, with some of the common attributes (ods, hds, etc. ) and validations.